src/conversion.rs

```diff
@@ -467,63 +408,71 @@ impl FromPy<()> for Py<PyTuple> {
 
 /// Raw level conversion between `*mut ffi::PyObject` and PyO3 types.
 pub unsafe trait FromPyPointer<'p>: Sized {
```

Comment:
There are so many methods in this trait. Is there a reason why we don't just have 

    unsafe fn from_owned_ptr(py: Python<'p>, ptr: *mut ffi::PyObject) -> PyResult<&'p Self>;
    unsafe fn from_borrowed_ptr(py: Python<'p>, ptr: *mut ffi::PyObject) -> PyResult<&'p Self>;

As the only trait method. And then just let users choose if they want to `.ok()`, `.unwrap()`, or `.expect()` this, instead of all the `_or_opt`, `_or_panic` etc.

This would be quite a big change, so maybe worth doing as a separate PR.

Reply:
Opened as #781